File: Python/cc20.py
"""🤬 Problem: Bleep Swear Words in a Sentence
Difficulty: Medium
Category: String Manipulation / Censorship

📝 Description:
Write a function that receives a string sentence and returns a version where any swear word is censored.

The list of swear words to censor includes (case-sensitive): ["fuck", "shit", "hell", "carajo"]
Each censored word must be transformed as follows:
Keep the first and last character
Replace all characters in between with asterisks *
Maintain the original capitalization of the first and last character
Punctuation attached to the word (like hell. or shit!) must remain at the end, uncensored.

🔧 Constraints:
1 ≤ sentence.length ≤ 1000
Sentence contains printable ASCII characters. Words are separated by spaces. 
Swear words may include ending punctuation (e.g. "shit!")
You may assume punctuation only appears at the end of words

🧠 Input Format:
A single string sentence

🧠 Output Format:
Return the sentence with swear words censored using the described rules.

🔍 Examples:

Example 1:
Input:                          Output:
"What the hell is this shit?"   "What the h**l is this s**t?"

Example 2:
Input:                          Output:
"FUCK that was intense!"        "F**K that was intense!"

Example 3:
Input:                          Output:
"oh shit! that’s bad"           "oh s**t! that’s bad"

Example 4:
Input:                          Output:
"nothing wrong here"            "nothing wrong here"

Example 5:
Input:                          Output:
"hell no, fuck this"            "h**l no, f**k this"
"""

#Pseudocode

#define a function wordBleep() passing 1 parameter sentence
# def wordBleep(sentence):
#   declare variables to store punctuation and to store modified sentence and set to empty string
    # censoredSentence = ""
    # punctuationToAddBack = ""
    # declare a list of all of the special characters that will remain unchanged
    # punctuationList = ["!", "?", ".", ","]
    # declaren list of all words that will be bleeped
    # wordToBleep = ["fuck", "shit", "hell", "carajo"]
    # declare a word variable to store the list of words after spliting sentence parameter
    # word = sentence.split(" ")
    # use for loop on word
    # for i in word:
        # use if to check if any of the words from the word list has a punctuation
        # if i in wordToBleep:
            # store punctuation in punctuationToAddBack
            # punctuationToAddBack = i[-1]
            # firstLttr = i[0]   
            # lastLttr = i[-1]
        # use if statement to handdle cassing
        # if word.lower() in wordToBleep:
            # word[0] + (len(word[1:-1]) * "*") + word[-1]
        # return censoredSentence


def wordBleep(sentence):
    censoredSentence = ""
    punctuationList = ["!", "?", ".", ","]
    punctuationToAddBack = ""
    wordToBleep = ["fuck", "shit", "hell", "carajo"]
    words = sentence.split(" ")
    for word in words:
        if word[-1] in punctuationList:
            punctuationToAddBack = word[-1]
            print(punctuationToAddBack, " - punctuation test")
            word = word[:-1]
            print(word, " - word to bleep test")
        if word.lower() in wordToBleep:
            word = word[0] + (len(word[1:-1])) * "*" + word[-1]
        print(word, " - test for word")
        censoredSentence += word + punctuationToAddBack + " "
        print(censoredSentence, " - final sentence test")
    return censoredSentence

print(wordBleep("What the hell is this shit?"))

File: JavaScript/cc20.js
/**
 * LeetCode-style Problem (Easy-Medium): Find Common Elements in Two Arrays
 *
 * Given two arrays of integers nums1 and nums2, return a new array containing 
 * all the unique numbers that appear in both arrays. The result can be returned 
 * in any order.
 *
 * Parameters:
 * - int[] nums1: A non-empty array of integers
 * - int[] nums2: A non-empty array of integers
 *
 * Returns:
 * - int[]: An array of integers that are common to both nums1 and nums2, with no duplicates
 *
 * Examples:
 * Input: nums1 = [1, 2, 2, 3], nums2 = [2, 3, 4]  
 * Output: [2, 3]
 *
 * Input: nums1 = [4, 5, 6], nums2 = [1, 2, 3]  
 * Output: []
 *
 * Input: nums1 = [1, 1, 1], nums2 = [1, 1]  
 * Output: [1]
 *
 * Constraints:
 * - 1 <= nums1.length, nums2.length <= 10^4
 * - -10^6 <= nums1[i], nums2[i] <= 10^6
 */

//create a function findingMatchingNumsInArray() passing 2 parameters nums1 and nums2
//function findingMatchingNumsInArray(nums1, nums2) {
//     declare a variable of set type
//     const matchesFound = new Set();
//     use a let of loop to iterate on nums1
//     for (let num of nums1) {
//          use an if statement to check for the condition (nums2, nums)
//          if (nums2, num) {
//              when true populate matchesFound using .push()
//                  matchesFound.push(num)
//                  const commonNumsInArrays = Array.from(matchesFound); 
//          }
//     }
//     return commonNumsInArrays;



function findingMatchingNumsInArray(nums1, nums2) {
    const matchesFound = new Set();
    for (let num of nums1) {
        if (nums2.includes(num)) {
            matchesFound.add(num)
        }
    }
    return Array.from(matchesFound);

}

console.log(findingMatchingNumsInArray([1, 1, 1], [1, 1]))